GitUI/CommandsDialogs/BuildReportTabPageExtension.cs

```diff
@@ -176,6 +180,25 @@ private void CreateBuildReportTabPage(TabControl tabControl)
             }
         }
 
+        private bool IsBuildResultPageEnabled()
+        {
+            var settings = GetModule().GetEffectiveSettings();
+
+            return settings.GetBool("BuildServer.ShowBuildResultPage") ?? true;
```

Comment:
Could you have used something like this?
https://github.com/gitextensions/gitextensions/blob/b210331fb67b2362cc1c666ad49911d4666d85e0/GitUI/BuildServerIntegration/BuildServerWatcher.cs#L288

Reply:
After posting and looking deeper into the code, I realise it is not possible in the current codebase.
I'll spend some time and will get back to you.

Reply:
I suggest to write the method as follows:
```
        private bool IsBuildResultPageEnabled()
        {
            var settings = GetModule().GetEffectiveSettings() as RepoDistSettings;
            return settings?.BuildServer.ShowBuildResultPage.ValueOrDefault ?? false;
        }
```

You're right the current architecture doesn't allow to easily get `BuildServer` config from an instance of `IGitModule`.
I've looked through the code and I can't really see an easy way out of it except for way you've done or the alternative I propose. 

Thoughts?

Reply:
Thanks for the reply.

As I think the best way is make settings avalible from IGitModule. But this requires refactoring.

Cast to RepoDistSettings is more simple and straightforward solution, but it is a downcast that violate the OOP principles.

Reply:
At times one needs to compromise.

Reply:
I'd prefer to keep the current code if you don't mind. Or should I use the approach you suggested?

Reply:
The trouble with `settings.GetBool("BuildServer.ShowBuildResultPage")` is that if the setting is renamed this code will break...

Reply:
Yes, you are right. I changed it. Rebase and squash it?